api4/plugin_test.go

```diff
@@ -767,6 +771,163 @@ func TestSearchGetMarketplacePlugins(t *testing.T) {
 		plugins, resp = th.SystemAdminClient.GetMarketplacePlugins(&model.MarketplacePluginFilter{Filter: "NOFILTER"})
 		CheckNoError(t, resp)
 		require.Nil(t, plugins)
+
+		// cleanup
+		ok, resp := th.SystemAdminClient.RemovePlugin(newPluginV1.Manifest.Id)
+		CheckNoError(t, resp)
+		assert.True(t, ok)
+
+		ok, resp = th.SystemAdminClient.RemovePlugin(newPluginV2.Manifest.Id)
+		CheckNoError(t, resp)
+		assert.True(t, ok)
+	})
+}
+
+func TestGetLocalPluginInMarketplace(t *testing.T) {
+	th := Setup().InitBasic()
+	defer th.TearDown()
+
+	samplePlugins := []*model.MarketplacePlugin{
+		{
+			BaseMarketplacePlugin: &model.BaseMarketplacePlugin{
+				HomepageURL: "https://example.com/mattermost/mattermost-plugin-nps",
+				IconData:    "https://example.com/icon.svg",
+				DownloadURL: "www.github.com/example",
+				Manifest: &model.Manifest{
+					Id:               "testplugin_v2",
+					Name:             "testplugin_v2",
+					Description:      "dsgsdg_v2",
+					Version:          "1.2.2",
+					MinServerVersion: "",
+				},
+			},
+			InstalledVersion: "",
+		},
+	}
+
+	testServer := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		res.WriteHeader(http.StatusOK)
+		json, err := json.Marshal([]*model.MarketplacePlugin{samplePlugins[0]})
+		require.NoError(t, err)
+		res.Write(json)
+	}))
+	defer testServer.Close()
+
+	th.App.UpdateConfig(func(cfg *model.Config) {
+		*cfg.PluginSettings.Enable = true
+		*cfg.PluginSettings.EnableMarketplace = true
+		*cfg.PluginSettings.MarketplaceUrl = testServer.URL
+	})
+
+	t.Run("Get plugins with EnableRemoteMarketplace enabled", func(t *testing.T) {
+
+		th.App.UpdateConfig(func(cfg *model.Config) {
+			*cfg.PluginSettings.EnableRemoteMarketplace = true
+		})
+
+		plugins, resp := th.SystemAdminClient.GetMarketplacePlugins(&model.MarketplacePluginFilter{})
+		CheckNoError(t, resp)
+
+		require.Equal(t, len(samplePlugins), len(plugins))
+		require.Equal(t, samplePlugins[0].Manifest.Id, plugins[0].Manifest.Id)
```

Comment:
better to assert the full plugin struct i.e. `require.Equal(t, samplePlugins[0], plugins[0])`